PWGMM/Mult/TableProducer/trackPropagation.cxx

```diff
@@ -185,6 +209,91 @@ struct AmbiguousTrackPropagation {
   }
 };
 
+struct AmbiguousTrackPropagationFwd {
+  Produces<aod::BestCollisionsFwd> fwdtracksBestCollisions;
+
+  using ExtBCs = soa::Join<aod::BCs, aod::Timestamps, aod::MatchedMulti>;
+
+  Service<o2::ccdb::BasicCCDBManager> ccdb;
+
+  int runNumber = -1;
+
+  o2::parameters::GRPMagField* grpmag = nullptr;
+  Configurable<std::string> grpmagPath{"grpmagPath", "GLO/Config/GRPMagField", "CCDB path of the GRPMagField object"};
+
+  float Bz = 0;
+
+  void initCCDB(ExtBCs::iterator const& bc)
+  {
+    if (runNumber == bc.runNumber()) {
+      return;
+    }
+    grpmag = ccdb->getForTimeStamp<o2::parameters::GRPMagField>(grpmagPath, bc.timestamp());
+    LOG(info) << "Setting magnetic field to current " << grpmag->getL3Current()
+              << " A for run " << bc.runNumber()
+              << " from its GRPMagField CCDB object";
+    o2::base::Propagator::initFieldFromGRP(grpmag);
+    runNumber = bc.runNumber();
+    o2::field::MagneticField* field = static_cast<o2::field::MagneticField*>(TGeoGlobalMagField::Instance()->GetField());
+    double centerMFT[3] = {0, 0, -61.4}; // Field at center of MFT
+    Bz = field->getBz(centerMFT);
+    printf("--------Bz = %f\n", Bz);
+  }
+
+  void process(aod::MFTTracks const&,
```

Comment:
This should probably have an option to exclude propagation of either central or forward ambiguous tracks when they are not needed.

Reply:
I agree,will that do with a different process name and a process switch ? If inside a struct I just do smthg like`if (doProcessCentral){return;}` will that work ? 
Is that what you meant ?

Reply:
I would suggest doing this similar to other service tasks, both tables are produced, but they are filled by process functions that can be disabled, so if the table is not needed it would be empty.

Reply:
Ok I did now :)